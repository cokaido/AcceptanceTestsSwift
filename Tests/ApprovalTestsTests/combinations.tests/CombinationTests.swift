import XCTest
#if os(OSX)
@testable import ApprovalTests_Swift
#elseif os(iOS)
@testable import ApprovalTests_iOS
#endif

class CombinationTests: XCTestCase {

    func threeParams(_ in1: Int, _ in2: Int, _ in3: Int ) -> Int {
        return in1 + in2 + in3
    }

    func twoIntParams(_ in1: Int, _ in2: Int ) -> Int {
        return in1 * in2
    }

    func twoFloatParams( _ in1: Float, _ in2: Float ) -> Float {
        return in1 * in2
    }

    func doubleInt(_ x: Int) -> Int {
        return 2 * x
    }

    func doubleFloat(_ x: Float) -> Float {
        return 2 * x
    }

    func threeMixedParams( _ anInt: Int, _ aFloat: Float, _ aString: String ) -> String {
        return String("\(anInt) \(aFloat) \(aString)")
    }

    func testCombinationsSingleParameterInt() {
        try! CombinationApprovals.verifyAllCombinations(doubleInt, [1,2,3,4,5])
    }

    func testCombinationsSingleParameterFloat() {
        try! CombinationApprovals.verifyAllCombinations(doubleFloat, [1.1, 2.2, 3.3, 4.4, 5.5])
    }

    func testCombinationsTwoParametersInt() {
        try! CombinationApprovals.verifyAllCombinations(twoIntParams, [1, 2], [3, 4])
    }

    func testCombinationsMixedParameters() {
        try! CombinationApprovals.verifyAllCombinations(threeMixedParams, [1,2,3], [4.4, 5.5], ["String1", "String2", "String3", "String4"])
    }
    
    func testCombinationsFourParameters() {
        let inCall: (Int, Int, Int, Int) -> String = {
            let sum = $0 + $1 + $2 + $3
            return String(sum)
        }

        try! CombinationApprovals.verifyAllCombinations(inCall, [1,2,3], [4,5], [6,7], [8,9])
    }

    func testCombinationsFiveParameters() {
        let inCall: (Int, Int, Int, Int, Int) -> String = {
            let sum = $0 + $1 + $2 + $3 + $4
            return String(sum)
        }
        
        try! CombinationApprovals.verifyAllCombinations(inCall, [1,2,3], [4,5], [6,7], [8,9], [10,11])
    }

    func testCombinationsSixParameters() {
        let inCall: (Int, Int, Int, Int, Int, Int) -> String = {
            let sum = $0 + $1 + $2 + $3 + $4 + $5
            return String(sum)
        }
        
        try! CombinationApprovals.verifyAllCombinations(inCall, [1,2,3], [4,5], [6,7], [8,9], [10,11], [12,13])
    }

    func testCombinationsSevenParameters() {
        let inCall: (Int, Int, Int, Int, Int, Int, Int) -> String = {
            let sum = $0 + $1 + $2 + $3 + $4 + $5 + $6
            return String(sum)
        }
        
        try! CombinationApprovals.verifyAllCombinations(inCall, [1,2,3], [4,5], [6,7], [8,9], [10,11], [12,13], [14,15])
    }

    func testCombinationsEightParameters() {
        let inCall: (Int, Int, Int, Int, Int, Int, Int, Int) -> String = {
            let sum = $0 + $1 + $2 + $3 + $4 + $5 + $6 + $7
            return String(sum)
        }
        
        try! CombinationApprovals.verifyAllCombinations(inCall, [1,2,3], [4,5], [6,7], [8,9], [10,11], [12,13], [14,15], [16,17])
    }

    func testCombinationsNineParameters() {
        let inCall: (Int, Int, Int, Int, Int, Int, Int, Int, Int) -> String = {
            let sum = $0 + $1 + $2 + $3 + $4 + $5 + $6 + $7 + $8
            return String(sum)
        }
        
        try! CombinationApprovals.verifyAllCombinations(inCall, [1,2,3], [4,5], [6,7], [8,9], [10,11], [12,13], [14,15], [16,17], [18,19])
    }

}
