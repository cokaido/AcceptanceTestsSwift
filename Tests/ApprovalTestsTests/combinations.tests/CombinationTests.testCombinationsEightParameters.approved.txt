[1, 4, 6, 8, 10, 12, 14, 16] -> 71
[1, 4, 6, 8, 10, 12, 14, 17] -> 72
[1, 4, 6, 8, 10, 12, 15, 16] -> 72
[1, 4, 6, 8, 10, 12, 15, 17] -> 73
[1, 4, 6, 8, 10, 13, 14, 16] -> 72
[1, 4, 6, 8, 10, 13, 14, 17] -> 73
[1, 4, 6, 8, 10, 13, 15, 16] -> 73
[1, 4, 6, 8, 10, 13, 15, 17] -> 74
[1, 4, 6, 8, 11, 12, 14, 16] -> 72
[1, 4, 6, 8, 11, 12, 14, 17] -> 73
[1, 4, 6, 8, 11, 12, 15, 16] -> 73
[1, 4, 6, 8, 11, 12, 15, 17] -> 74
[1, 4, 6, 8, 11, 13, 14, 16] -> 73
[1, 4, 6, 8, 11, 13, 14, 17] -> 74
[1, 4, 6, 8, 11, 13, 15, 16] -> 74
[1, 4, 6, 8, 11, 13, 15, 17] -> 75
[1, 4, 6, 9, 10, 12, 14, 16] -> 72
[1, 4, 6, 9, 10, 12, 14, 17] -> 73
[1, 4, 6, 9, 10, 12, 15, 16] -> 73
[1, 4, 6, 9, 10, 12, 15, 17] -> 74
[1, 4, 6, 9, 10, 13, 14, 16] -> 73
[1, 4, 6, 9, 10, 13, 14, 17] -> 74
[1, 4, 6, 9, 10, 13, 15, 16] -> 74
[1, 4, 6, 9, 10, 13, 15, 17] -> 75
[1, 4, 6, 9, 11, 12, 14, 16] -> 73
[1, 4, 6, 9, 11, 12, 14, 17] -> 74
[1, 4, 6, 9, 11, 12, 15, 16] -> 74
[1, 4, 6, 9, 11, 12, 15, 17] -> 75
[1, 4, 6, 9, 11, 13, 14, 16] -> 74
[1, 4, 6, 9, 11, 13, 14, 17] -> 75
[1, 4, 6, 9, 11, 13, 15, 16] -> 75
[1, 4, 6, 9, 11, 13, 15, 17] -> 76
[1, 4, 7, 8, 10, 12, 14, 16] -> 72
[1, 4, 7, 8, 10, 12, 14, 17] -> 73
[1, 4, 7, 8, 10, 12, 15, 16] -> 73
[1, 4, 7, 8, 10, 12, 15, 17] -> 74
[1, 4, 7, 8, 10, 13, 14, 16] -> 73
[1, 4, 7, 8, 10, 13, 14, 17] -> 74
[1, 4, 7, 8, 10, 13, 15, 16] -> 74
[1, 4, 7, 8, 10, 13, 15, 17] -> 75
[1, 4, 7, 8, 11, 12, 14, 16] -> 73
[1, 4, 7, 8, 11, 12, 14, 17] -> 74
[1, 4, 7, 8, 11, 12, 15, 16] -> 74
[1, 4, 7, 8, 11, 12, 15, 17] -> 75
[1, 4, 7, 8, 11, 13, 14, 16] -> 74
[1, 4, 7, 8, 11, 13, 14, 17] -> 75
[1, 4, 7, 8, 11, 13, 15, 16] -> 75
[1, 4, 7, 8, 11, 13, 15, 17] -> 76
[1, 4, 7, 9, 10, 12, 14, 16] -> 73
[1, 4, 7, 9, 10, 12, 14, 17] -> 74
[1, 4, 7, 9, 10, 12, 15, 16] -> 74
[1, 4, 7, 9, 10, 12, 15, 17] -> 75
[1, 4, 7, 9, 10, 13, 14, 16] -> 74
[1, 4, 7, 9, 10, 13, 14, 17] -> 75
[1, 4, 7, 9, 10, 13, 15, 16] -> 75
[1, 4, 7, 9, 10, 13, 15, 17] -> 76
[1, 4, 7, 9, 11, 12, 14, 16] -> 74
[1, 4, 7, 9, 11, 12, 14, 17] -> 75
[1, 4, 7, 9, 11, 12, 15, 16] -> 75
[1, 4, 7, 9, 11, 12, 15, 17] -> 76
[1, 4, 7, 9, 11, 13, 14, 16] -> 75
[1, 4, 7, 9, 11, 13, 14, 17] -> 76
[1, 4, 7, 9, 11, 13, 15, 16] -> 76
[1, 4, 7, 9, 11, 13, 15, 17] -> 77
[1, 5, 6, 8, 10, 12, 14, 16] -> 72
[1, 5, 6, 8, 10, 12, 14, 17] -> 73
[1, 5, 6, 8, 10, 12, 15, 16] -> 73
[1, 5, 6, 8, 10, 12, 15, 17] -> 74
[1, 5, 6, 8, 10, 13, 14, 16] -> 73
[1, 5, 6, 8, 10, 13, 14, 17] -> 74
[1, 5, 6, 8, 10, 13, 15, 16] -> 74
[1, 5, 6, 8, 10, 13, 15, 17] -> 75
[1, 5, 6, 8, 11, 12, 14, 16] -> 73
[1, 5, 6, 8, 11, 12, 14, 17] -> 74
[1, 5, 6, 8, 11, 12, 15, 16] -> 74
[1, 5, 6, 8, 11, 12, 15, 17] -> 75
[1, 5, 6, 8, 11, 13, 14, 16] -> 74
[1, 5, 6, 8, 11, 13, 14, 17] -> 75
[1, 5, 6, 8, 11, 13, 15, 16] -> 75
[1, 5, 6, 8, 11, 13, 15, 17] -> 76
[1, 5, 6, 9, 10, 12, 14, 16] -> 73
[1, 5, 6, 9, 10, 12, 14, 17] -> 74
[1, 5, 6, 9, 10, 12, 15, 16] -> 74
[1, 5, 6, 9, 10, 12, 15, 17] -> 75
[1, 5, 6, 9, 10, 13, 14, 16] -> 74
[1, 5, 6, 9, 10, 13, 14, 17] -> 75
[1, 5, 6, 9, 10, 13, 15, 16] -> 75
[1, 5, 6, 9, 10, 13, 15, 17] -> 76
[1, 5, 6, 9, 11, 12, 14, 16] -> 74
[1, 5, 6, 9, 11, 12, 14, 17] -> 75
[1, 5, 6, 9, 11, 12, 15, 16] -> 75
[1, 5, 6, 9, 11, 12, 15, 17] -> 76
[1, 5, 6, 9, 11, 13, 14, 16] -> 75
[1, 5, 6, 9, 11, 13, 14, 17] -> 76
[1, 5, 6, 9, 11, 13, 15, 16] -> 76
[1, 5, 6, 9, 11, 13, 15, 17] -> 77
[1, 5, 7, 8, 10, 12, 14, 16] -> 73
[1, 5, 7, 8, 10, 12, 14, 17] -> 74
[1, 5, 7, 8, 10, 12, 15, 16] -> 74
[1, 5, 7, 8, 10, 12, 15, 17] -> 75
[1, 5, 7, 8, 10, 13, 14, 16] -> 74
[1, 5, 7, 8, 10, 13, 14, 17] -> 75
[1, 5, 7, 8, 10, 13, 15, 16] -> 75
[1, 5, 7, 8, 10, 13, 15, 17] -> 76
[1, 5, 7, 8, 11, 12, 14, 16] -> 74
[1, 5, 7, 8, 11, 12, 14, 17] -> 75
[1, 5, 7, 8, 11, 12, 15, 16] -> 75
[1, 5, 7, 8, 11, 12, 15, 17] -> 76
[1, 5, 7, 8, 11, 13, 14, 16] -> 75
[1, 5, 7, 8, 11, 13, 14, 17] -> 76
[1, 5, 7, 8, 11, 13, 15, 16] -> 76
[1, 5, 7, 8, 11, 13, 15, 17] -> 77
[1, 5, 7, 9, 10, 12, 14, 16] -> 74
[1, 5, 7, 9, 10, 12, 14, 17] -> 75
[1, 5, 7, 9, 10, 12, 15, 16] -> 75
[1, 5, 7, 9, 10, 12, 15, 17] -> 76
[1, 5, 7, 9, 10, 13, 14, 16] -> 75
[1, 5, 7, 9, 10, 13, 14, 17] -> 76
[1, 5, 7, 9, 10, 13, 15, 16] -> 76
[1, 5, 7, 9, 10, 13, 15, 17] -> 77
[1, 5, 7, 9, 11, 12, 14, 16] -> 75
[1, 5, 7, 9, 11, 12, 14, 17] -> 76
[1, 5, 7, 9, 11, 12, 15, 16] -> 76
[1, 5, 7, 9, 11, 12, 15, 17] -> 77
[1, 5, 7, 9, 11, 13, 14, 16] -> 76
[1, 5, 7, 9, 11, 13, 14, 17] -> 77
[1, 5, 7, 9, 11, 13, 15, 16] -> 77
[1, 5, 7, 9, 11, 13, 15, 17] -> 78
[2, 4, 6, 8, 10, 12, 14, 16] -> 72
[2, 4, 6, 8, 10, 12, 14, 17] -> 73
[2, 4, 6, 8, 10, 12, 15, 16] -> 73
[2, 4, 6, 8, 10, 12, 15, 17] -> 74
[2, 4, 6, 8, 10, 13, 14, 16] -> 73
[2, 4, 6, 8, 10, 13, 14, 17] -> 74
[2, 4, 6, 8, 10, 13, 15, 16] -> 74
[2, 4, 6, 8, 10, 13, 15, 17] -> 75
[2, 4, 6, 8, 11, 12, 14, 16] -> 73
[2, 4, 6, 8, 11, 12, 14, 17] -> 74
[2, 4, 6, 8, 11, 12, 15, 16] -> 74
[2, 4, 6, 8, 11, 12, 15, 17] -> 75
[2, 4, 6, 8, 11, 13, 14, 16] -> 74
[2, 4, 6, 8, 11, 13, 14, 17] -> 75
[2, 4, 6, 8, 11, 13, 15, 16] -> 75
[2, 4, 6, 8, 11, 13, 15, 17] -> 76
[2, 4, 6, 9, 10, 12, 14, 16] -> 73
[2, 4, 6, 9, 10, 12, 14, 17] -> 74
[2, 4, 6, 9, 10, 12, 15, 16] -> 74
[2, 4, 6, 9, 10, 12, 15, 17] -> 75
[2, 4, 6, 9, 10, 13, 14, 16] -> 74
[2, 4, 6, 9, 10, 13, 14, 17] -> 75
[2, 4, 6, 9, 10, 13, 15, 16] -> 75
[2, 4, 6, 9, 10, 13, 15, 17] -> 76
[2, 4, 6, 9, 11, 12, 14, 16] -> 74
[2, 4, 6, 9, 11, 12, 14, 17] -> 75
[2, 4, 6, 9, 11, 12, 15, 16] -> 75
[2, 4, 6, 9, 11, 12, 15, 17] -> 76
[2, 4, 6, 9, 11, 13, 14, 16] -> 75
[2, 4, 6, 9, 11, 13, 14, 17] -> 76
[2, 4, 6, 9, 11, 13, 15, 16] -> 76
[2, 4, 6, 9, 11, 13, 15, 17] -> 77
[2, 4, 7, 8, 10, 12, 14, 16] -> 73
[2, 4, 7, 8, 10, 12, 14, 17] -> 74
[2, 4, 7, 8, 10, 12, 15, 16] -> 74
[2, 4, 7, 8, 10, 12, 15, 17] -> 75
[2, 4, 7, 8, 10, 13, 14, 16] -> 74
[2, 4, 7, 8, 10, 13, 14, 17] -> 75
[2, 4, 7, 8, 10, 13, 15, 16] -> 75
[2, 4, 7, 8, 10, 13, 15, 17] -> 76
[2, 4, 7, 8, 11, 12, 14, 16] -> 74
[2, 4, 7, 8, 11, 12, 14, 17] -> 75
[2, 4, 7, 8, 11, 12, 15, 16] -> 75
[2, 4, 7, 8, 11, 12, 15, 17] -> 76
[2, 4, 7, 8, 11, 13, 14, 16] -> 75
[2, 4, 7, 8, 11, 13, 14, 17] -> 76
[2, 4, 7, 8, 11, 13, 15, 16] -> 76
[2, 4, 7, 8, 11, 13, 15, 17] -> 77
[2, 4, 7, 9, 10, 12, 14, 16] -> 74
[2, 4, 7, 9, 10, 12, 14, 17] -> 75
[2, 4, 7, 9, 10, 12, 15, 16] -> 75
[2, 4, 7, 9, 10, 12, 15, 17] -> 76
[2, 4, 7, 9, 10, 13, 14, 16] -> 75
[2, 4, 7, 9, 10, 13, 14, 17] -> 76
[2, 4, 7, 9, 10, 13, 15, 16] -> 76
[2, 4, 7, 9, 10, 13, 15, 17] -> 77
[2, 4, 7, 9, 11, 12, 14, 16] -> 75
[2, 4, 7, 9, 11, 12, 14, 17] -> 76
[2, 4, 7, 9, 11, 12, 15, 16] -> 76
[2, 4, 7, 9, 11, 12, 15, 17] -> 77
[2, 4, 7, 9, 11, 13, 14, 16] -> 76
[2, 4, 7, 9, 11, 13, 14, 17] -> 77
[2, 4, 7, 9, 11, 13, 15, 16] -> 77
[2, 4, 7, 9, 11, 13, 15, 17] -> 78
[2, 5, 6, 8, 10, 12, 14, 16] -> 73
[2, 5, 6, 8, 10, 12, 14, 17] -> 74
[2, 5, 6, 8, 10, 12, 15, 16] -> 74
[2, 5, 6, 8, 10, 12, 15, 17] -> 75
[2, 5, 6, 8, 10, 13, 14, 16] -> 74
[2, 5, 6, 8, 10, 13, 14, 17] -> 75
[2, 5, 6, 8, 10, 13, 15, 16] -> 75
[2, 5, 6, 8, 10, 13, 15, 17] -> 76
[2, 5, 6, 8, 11, 12, 14, 16] -> 74
[2, 5, 6, 8, 11, 12, 14, 17] -> 75
[2, 5, 6, 8, 11, 12, 15, 16] -> 75
[2, 5, 6, 8, 11, 12, 15, 17] -> 76
[2, 5, 6, 8, 11, 13, 14, 16] -> 75
[2, 5, 6, 8, 11, 13, 14, 17] -> 76
[2, 5, 6, 8, 11, 13, 15, 16] -> 76
[2, 5, 6, 8, 11, 13, 15, 17] -> 77
[2, 5, 6, 9, 10, 12, 14, 16] -> 74
[2, 5, 6, 9, 10, 12, 14, 17] -> 75
[2, 5, 6, 9, 10, 12, 15, 16] -> 75
[2, 5, 6, 9, 10, 12, 15, 17] -> 76
[2, 5, 6, 9, 10, 13, 14, 16] -> 75
[2, 5, 6, 9, 10, 13, 14, 17] -> 76
[2, 5, 6, 9, 10, 13, 15, 16] -> 76
[2, 5, 6, 9, 10, 13, 15, 17] -> 77
[2, 5, 6, 9, 11, 12, 14, 16] -> 75
[2, 5, 6, 9, 11, 12, 14, 17] -> 76
[2, 5, 6, 9, 11, 12, 15, 16] -> 76
[2, 5, 6, 9, 11, 12, 15, 17] -> 77
[2, 5, 6, 9, 11, 13, 14, 16] -> 76
[2, 5, 6, 9, 11, 13, 14, 17] -> 77
[2, 5, 6, 9, 11, 13, 15, 16] -> 77
[2, 5, 6, 9, 11, 13, 15, 17] -> 78
[2, 5, 7, 8, 10, 12, 14, 16] -> 74
[2, 5, 7, 8, 10, 12, 14, 17] -> 75
[2, 5, 7, 8, 10, 12, 15, 16] -> 75
[2, 5, 7, 8, 10, 12, 15, 17] -> 76
[2, 5, 7, 8, 10, 13, 14, 16] -> 75
[2, 5, 7, 8, 10, 13, 14, 17] -> 76
[2, 5, 7, 8, 10, 13, 15, 16] -> 76
[2, 5, 7, 8, 10, 13, 15, 17] -> 77
[2, 5, 7, 8, 11, 12, 14, 16] -> 75
[2, 5, 7, 8, 11, 12, 14, 17] -> 76
[2, 5, 7, 8, 11, 12, 15, 16] -> 76
[2, 5, 7, 8, 11, 12, 15, 17] -> 77
[2, 5, 7, 8, 11, 13, 14, 16] -> 76
[2, 5, 7, 8, 11, 13, 14, 17] -> 77
[2, 5, 7, 8, 11, 13, 15, 16] -> 77
[2, 5, 7, 8, 11, 13, 15, 17] -> 78
[2, 5, 7, 9, 10, 12, 14, 16] -> 75
[2, 5, 7, 9, 10, 12, 14, 17] -> 76
[2, 5, 7, 9, 10, 12, 15, 16] -> 76
[2, 5, 7, 9, 10, 12, 15, 17] -> 77
[2, 5, 7, 9, 10, 13, 14, 16] -> 76
[2, 5, 7, 9, 10, 13, 14, 17] -> 77
[2, 5, 7, 9, 10, 13, 15, 16] -> 77
[2, 5, 7, 9, 10, 13, 15, 17] -> 78
[2, 5, 7, 9, 11, 12, 14, 16] -> 76
[2, 5, 7, 9, 11, 12, 14, 17] -> 77
[2, 5, 7, 9, 11, 12, 15, 16] -> 77
[2, 5, 7, 9, 11, 12, 15, 17] -> 78
[2, 5, 7, 9, 11, 13, 14, 16] -> 77
[2, 5, 7, 9, 11, 13, 14, 17] -> 78
[2, 5, 7, 9, 11, 13, 15, 16] -> 78
[2, 5, 7, 9, 11, 13, 15, 17] -> 79
[3, 4, 6, 8, 10, 12, 14, 16] -> 73
[3, 4, 6, 8, 10, 12, 14, 17] -> 74
[3, 4, 6, 8, 10, 12, 15, 16] -> 74
[3, 4, 6, 8, 10, 12, 15, 17] -> 75
[3, 4, 6, 8, 10, 13, 14, 16] -> 74
[3, 4, 6, 8, 10, 13, 14, 17] -> 75
[3, 4, 6, 8, 10, 13, 15, 16] -> 75
[3, 4, 6, 8, 10, 13, 15, 17] -> 76
[3, 4, 6, 8, 11, 12, 14, 16] -> 74
[3, 4, 6, 8, 11, 12, 14, 17] -> 75
[3, 4, 6, 8, 11, 12, 15, 16] -> 75
[3, 4, 6, 8, 11, 12, 15, 17] -> 76
[3, 4, 6, 8, 11, 13, 14, 16] -> 75
[3, 4, 6, 8, 11, 13, 14, 17] -> 76
[3, 4, 6, 8, 11, 13, 15, 16] -> 76
[3, 4, 6, 8, 11, 13, 15, 17] -> 77
[3, 4, 6, 9, 10, 12, 14, 16] -> 74
[3, 4, 6, 9, 10, 12, 14, 17] -> 75
[3, 4, 6, 9, 10, 12, 15, 16] -> 75
[3, 4, 6, 9, 10, 12, 15, 17] -> 76
[3, 4, 6, 9, 10, 13, 14, 16] -> 75
[3, 4, 6, 9, 10, 13, 14, 17] -> 76
[3, 4, 6, 9, 10, 13, 15, 16] -> 76
[3, 4, 6, 9, 10, 13, 15, 17] -> 77
[3, 4, 6, 9, 11, 12, 14, 16] -> 75
[3, 4, 6, 9, 11, 12, 14, 17] -> 76
[3, 4, 6, 9, 11, 12, 15, 16] -> 76
[3, 4, 6, 9, 11, 12, 15, 17] -> 77
[3, 4, 6, 9, 11, 13, 14, 16] -> 76
[3, 4, 6, 9, 11, 13, 14, 17] -> 77
[3, 4, 6, 9, 11, 13, 15, 16] -> 77
[3, 4, 6, 9, 11, 13, 15, 17] -> 78
[3, 4, 7, 8, 10, 12, 14, 16] -> 74
[3, 4, 7, 8, 10, 12, 14, 17] -> 75
[3, 4, 7, 8, 10, 12, 15, 16] -> 75
[3, 4, 7, 8, 10, 12, 15, 17] -> 76
[3, 4, 7, 8, 10, 13, 14, 16] -> 75
[3, 4, 7, 8, 10, 13, 14, 17] -> 76
[3, 4, 7, 8, 10, 13, 15, 16] -> 76
[3, 4, 7, 8, 10, 13, 15, 17] -> 77
[3, 4, 7, 8, 11, 12, 14, 16] -> 75
[3, 4, 7, 8, 11, 12, 14, 17] -> 76
[3, 4, 7, 8, 11, 12, 15, 16] -> 76
[3, 4, 7, 8, 11, 12, 15, 17] -> 77
[3, 4, 7, 8, 11, 13, 14, 16] -> 76
[3, 4, 7, 8, 11, 13, 14, 17] -> 77
[3, 4, 7, 8, 11, 13, 15, 16] -> 77
[3, 4, 7, 8, 11, 13, 15, 17] -> 78
[3, 4, 7, 9, 10, 12, 14, 16] -> 75
[3, 4, 7, 9, 10, 12, 14, 17] -> 76
[3, 4, 7, 9, 10, 12, 15, 16] -> 76
[3, 4, 7, 9, 10, 12, 15, 17] -> 77
[3, 4, 7, 9, 10, 13, 14, 16] -> 76
[3, 4, 7, 9, 10, 13, 14, 17] -> 77
[3, 4, 7, 9, 10, 13, 15, 16] -> 77
[3, 4, 7, 9, 10, 13, 15, 17] -> 78
[3, 4, 7, 9, 11, 12, 14, 16] -> 76
[3, 4, 7, 9, 11, 12, 14, 17] -> 77
[3, 4, 7, 9, 11, 12, 15, 16] -> 77
[3, 4, 7, 9, 11, 12, 15, 17] -> 78
[3, 4, 7, 9, 11, 13, 14, 16] -> 77
[3, 4, 7, 9, 11, 13, 14, 17] -> 78
[3, 4, 7, 9, 11, 13, 15, 16] -> 78
[3, 4, 7, 9, 11, 13, 15, 17] -> 79
[3, 5, 6, 8, 10, 12, 14, 16] -> 74
[3, 5, 6, 8, 10, 12, 14, 17] -> 75
[3, 5, 6, 8, 10, 12, 15, 16] -> 75
[3, 5, 6, 8, 10, 12, 15, 17] -> 76
[3, 5, 6, 8, 10, 13, 14, 16] -> 75
[3, 5, 6, 8, 10, 13, 14, 17] -> 76
[3, 5, 6, 8, 10, 13, 15, 16] -> 76
[3, 5, 6, 8, 10, 13, 15, 17] -> 77
[3, 5, 6, 8, 11, 12, 14, 16] -> 75
[3, 5, 6, 8, 11, 12, 14, 17] -> 76
[3, 5, 6, 8, 11, 12, 15, 16] -> 76
[3, 5, 6, 8, 11, 12, 15, 17] -> 77
[3, 5, 6, 8, 11, 13, 14, 16] -> 76
[3, 5, 6, 8, 11, 13, 14, 17] -> 77
[3, 5, 6, 8, 11, 13, 15, 16] -> 77
[3, 5, 6, 8, 11, 13, 15, 17] -> 78
[3, 5, 6, 9, 10, 12, 14, 16] -> 75
[3, 5, 6, 9, 10, 12, 14, 17] -> 76
[3, 5, 6, 9, 10, 12, 15, 16] -> 76
[3, 5, 6, 9, 10, 12, 15, 17] -> 77
[3, 5, 6, 9, 10, 13, 14, 16] -> 76
[3, 5, 6, 9, 10, 13, 14, 17] -> 77
[3, 5, 6, 9, 10, 13, 15, 16] -> 77
[3, 5, 6, 9, 10, 13, 15, 17] -> 78
[3, 5, 6, 9, 11, 12, 14, 16] -> 76
[3, 5, 6, 9, 11, 12, 14, 17] -> 77
[3, 5, 6, 9, 11, 12, 15, 16] -> 77
[3, 5, 6, 9, 11, 12, 15, 17] -> 78
[3, 5, 6, 9, 11, 13, 14, 16] -> 77
[3, 5, 6, 9, 11, 13, 14, 17] -> 78
[3, 5, 6, 9, 11, 13, 15, 16] -> 78
[3, 5, 6, 9, 11, 13, 15, 17] -> 79
[3, 5, 7, 8, 10, 12, 14, 16] -> 75
[3, 5, 7, 8, 10, 12, 14, 17] -> 76
[3, 5, 7, 8, 10, 12, 15, 16] -> 76
[3, 5, 7, 8, 10, 12, 15, 17] -> 77
[3, 5, 7, 8, 10, 13, 14, 16] -> 76
[3, 5, 7, 8, 10, 13, 14, 17] -> 77
[3, 5, 7, 8, 10, 13, 15, 16] -> 77
[3, 5, 7, 8, 10, 13, 15, 17] -> 78
[3, 5, 7, 8, 11, 12, 14, 16] -> 76
[3, 5, 7, 8, 11, 12, 14, 17] -> 77
[3, 5, 7, 8, 11, 12, 15, 16] -> 77
[3, 5, 7, 8, 11, 12, 15, 17] -> 78
[3, 5, 7, 8, 11, 13, 14, 16] -> 77
[3, 5, 7, 8, 11, 13, 14, 17] -> 78
[3, 5, 7, 8, 11, 13, 15, 16] -> 78
[3, 5, 7, 8, 11, 13, 15, 17] -> 79
[3, 5, 7, 9, 10, 12, 14, 16] -> 76
[3, 5, 7, 9, 10, 12, 14, 17] -> 77
[3, 5, 7, 9, 10, 12, 15, 16] -> 77
[3, 5, 7, 9, 10, 12, 15, 17] -> 78
[3, 5, 7, 9, 10, 13, 14, 16] -> 77
[3, 5, 7, 9, 10, 13, 14, 17] -> 78
[3, 5, 7, 9, 10, 13, 15, 16] -> 78
[3, 5, 7, 9, 10, 13, 15, 17] -> 79
[3, 5, 7, 9, 11, 12, 14, 16] -> 77
[3, 5, 7, 9, 11, 12, 14, 17] -> 78
[3, 5, 7, 9, 11, 12, 15, 16] -> 78
[3, 5, 7, 9, 11, 12, 15, 17] -> 79
[3, 5, 7, 9, 11, 13, 14, 16] -> 78
[3, 5, 7, 9, 11, 13, 14, 17] -> 79
[3, 5, 7, 9, 11, 13, 15, 16] -> 79
[3, 5, 7, 9, 11, 13, 15, 17] -> 80
